legend-engine-xts-flatdata/legend-engine-xt-flatdata-model/src/test/java/org/finos/legend/engine/external/format/flatdata/grammar/driver/TestFixedWidth.java

```diff
@@ -115,6 +118,124 @@ public void canReadErrorFree()
+                AbstractDriverTest.rValue("6:8", "1.1"),
+                AbstractDriverTest.rValue("9:12", "Alex"),
+                AbstractDriverTest.rValue("13:22", "2013-08-13"),
+                AbstractDriverTest.rValue("23:EOL", "Mrs")
+        );
+
+        List<ExpectedRecordValue> expectedRecordValues2 = Arrays.asList(
+                AbstractDriverTest.rValue("1:2", "GG"),
+                AbstractDriverTest.rValue("3:4", "26"),
+                AbstractDriverTest.rValue("5:5", "0"),
+                AbstractDriverTest.rValue("6:8", "1.2"),
+                AbstractDriverTest.rValue("9:12", "Brad"),
+                AbstractDriverTest.rValue("13:22", "2003-01-01"),
+                AbstractDriverTest.rValue("23:EOL", "Mr")
+        );
+
+        List<AbstractDriverTest.ExpectedRecordValue> expectedRecordValues3 = Arrays.asList(
+                AbstractDriverTest.rValue("1:2", "FB"),
+                AbstractDriverTest.rValue("3:4", "27"),
+                AbstractDriverTest.rValue("5:5", "1"),
+                AbstractDriverTest.rValue("6:8", "1.3"),
+                AbstractDriverTest.rValue("9:12", "Karl"),
+                AbstractDriverTest.rValue("13:22", "2011-11-26"),
+                AbstractDriverTest.rValue("23:EOL", "Master")
+        );
+
+        Assert.assertTrue(new FixedWidthDriverDescription().validate(flatData, flatData.sections.get(0)).isEmpty());
+        assertSource(1, 1, "GS2511.1Alex2013-08-13Mrs", expectedRecordValues1, records.get(0));
+        Person p1 = records.get(0).getValue();
+        Assert.assertEquals("GS", p1.FIRM);
+        Assert.assertEquals(25L, p1.AGE);
+        Assert.assertEquals(1, p1.MASTER);
+        Assert.assertEquals("1.1", p1.WEIGHT);
+        Assert.assertEquals("Alex", p1.NAME);
+        Assert.assertEquals(LocalDate.parse("2013-08-13"), p1.EMPLOYED_DATE);
+        Assert.assertEquals("Mrs", p1.TITLE);
+
+        assertSource(2, 2, "GG2601.2Brad2003-01-01Mr", expectedRecordValues2, records.get(1));
+        Person p2 = records.get(1).getValue();
+        Assert.assertEquals("GG", p2.FIRM);
+        Assert.assertEquals(26L, p2.AGE);
+        Assert.assertEquals(0, p2.MASTER);
+        Assert.assertEquals("1.2", p2.WEIGHT);
+        Assert.assertEquals("Brad", p2.NAME);
+        Assert.assertEquals(LocalDate.parse("2003-01-01"), p2.EMPLOYED_DATE);
+        Assert.assertEquals("Mr", p2.TITLE);
+
+
+        assertSource(3, 3, "FB2711.3Karl2011-11-26Master", expectedRecordValues3, records.get(2));
+        Person p3 = records.get(2).getValue();
+        Assert.assertEquals("FB", p3.FIRM);
+        Assert.assertEquals(27L, p3.AGE);
+        Assert.assertEquals(1, p3.MASTER);
+        Assert.assertEquals("1.3", p3.WEIGHT);
+        Assert.assertEquals("Karl", p3.NAME);
+        Assert.assertEquals(LocalDate.parse("2011-11-26"), p3.EMPLOYED_DATE);
+        Assert.assertEquals("Master", p3.TITLE);
+    }
+
+    @Test
+    public void checkValidationErrorIncorrectFinalColumn()
```

Comment:
Similar to this, can we add a test where EOL appears before the last field to validate the exception behavior?

Reply:
Yes - added a test for this